addons/mail/static/src/core/common/composer.js

```diff
@@ -639,4 +664,24 @@ export class Composer extends Component {
             this.threadService.markAsRead(this.props.composer.thread);
         }
     }
+
+    saveInputToLocalStorage() {
+        const fullComposerContent =
+            document
+                .querySelector(".o_mail_composer_form_view #body_0")
+                ?.innerText.replace(/(\t|\n)+/g, "\n") ?? this.props.composer.textInputContent;
+        browser.localStorage.setItem(
+            `${this.thread.localId}_fullComposerText`,
+            fullComposerContent
+        );
+    }
+
+    setInputFromLocalStorage() {
+        const fullComposerContent = browser.localStorage.getItem(
+            `${this.thread.localId}_fullComposerText`
+        );
+        if (fullComposerContent) {
+            this.props.composer.textInputContent = fullComposerContent;
+        }
+    }
```

Comment:
I would store `composer.localId` rather than `thread.localId`, so that it works for all composers including editing a message. And it's conceptually more immediate on composer.

Suffix `_fullComposerText` is not necessary. Let's just use composer local id for entry :)